src/js/util/param_spec.js

```diff
@@ -0,0 +1,40 @@
+function ParamSpec(sourceLow, sourceHigh, distLow, distHigh, clamp = true) {
+  const domain = sourceHigh - sourceLow
+  const range = distHigh - distLow
+  let map
+  let unmap
+
+  if (clamp) {
```

Comment:
if asked to map a value out of range it restricts to the min/max of the output range